cli/src/lib/plugin/webpack.config.js

```diff
@@ -0,0 +1,255 @@
+                        },
+                    },
+                }),
+                new CssMinimizerPlugin(),
+            ],
+            splitChunks: {
+                chunks: 'all',
+            },
+        },
+        plugins: [
+            isDevelopment && new ReactRefreshWebpackPlugin(),
+            new HtmlWebpackPlugin(
+                Object.assign(
+                    {
+                        inject: true,
+                        filename: 'plugin.html',
+                        template: paths.shellPublicPluginHtml,
+                    },
+                    isProduction
+                        ? {
+                              minify: {
+                                  removeComments: true,
+                                  collapseWhitespace: true,
+                                  removeRedundantAttributes: true,
+                                  useShortDoctype: true,
+                                  removeEmptyAttributes: true,
+                                  removeStyleLinkTypeAttributes: true,
+                                  keepClosingSlash: true,
+                                  minifyJS: true,
+                                  minifyCSS: true,
+                                  minifyURLs: true,
+                              },
+                          }
+                        : undefined
+                )
+            ),
+            isProduction &&
+                new MiniCssExtractPlugin({
+                    filename: 'static/css/[name].[contenthash:8].css',
+                    chunkFilename:
+                        'static/css/[name].[contenthash:8].chunk.css',
+                }),
+            // Makes some environment variables available to the JS code, for example:
+            // if (process.env.NODE_ENV === 'production') { ... }.
+            // It is absolutely essential that NODE_ENV is set to production
+            // during a production build.
+            // Otherwise React will be compiled in the very slow development mode.
+            new webpack.DefinePlugin({
+                'process.env': {
+                    ...Object.keys(shellEnv).reduce((env, key) => {
+                        env[key] = JSON.stringify(shellEnv[key])
+                        return env
+                    }, {}),
+                    NODE_ENV: JSON.stringify(webpackEnv),
+                },
+            }),
+            new webpack.IgnorePlugin(/^\.\/locale$/, /moment$/),
+        ].filter(Boolean),
+        module: {
+            rules: [
+                {
+                    oneOf: [
+                        {
+                            test: /\.(js|mjs|jsx|ts|tsx)$/,
+                            include: paths.shellSrc,
+                            use: {
+                                loader: require.resolve('babel-loader'),
+                                options: {
+                                    ...makeBabelConfig({ mode: webpackEnv }),
```

Comment:
This webpack config could also be used for apps, eliminating the need for a separate babel transformation pass.

Reply:
This would also allow for code-splitting between app and plugin, which would be advantageous